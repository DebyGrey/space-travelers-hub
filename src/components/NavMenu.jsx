import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavMenu = ({ path, name }) => (
  <li className="last:border-l h-auto py-0 border-gray-400 px-2">
    <NavLink
      to={path}
      className={({ isActive }) => (isActive ? 'text-blue-600 underline' : 'text-blue-600')}
    >
      {name}
    </NavLink>
  </li>
);
export default NavMenu;

NavMenu.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};
