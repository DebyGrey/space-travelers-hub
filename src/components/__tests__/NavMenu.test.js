import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import NavMenu from '../NavMenu';

// Mock PropTypes validation
jest.spyOn(PropTypes, 'string').mockImplementation(() => null);

describe('NavMenu', () => {
  test('renders the NavMenu component with the provided props', () => {
    const path = '/';
    const name = 'Home';

    render(
      <MemoryRouter>
        <NavMenu path={path} name={name} />
      </MemoryRouter>,
    );

    const navLinkElement = screen.getByText(name);
    expect(navLinkElement).toBeInTheDocument();
    expect(navLinkElement).toHaveAttribute('href', path);
  });

  test('matches snapshot', () => {
    const path = '/';
    const name = 'Home';

    const { asFragment } = render(
      <MemoryRouter>
        <NavMenu path={path} name={name} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
