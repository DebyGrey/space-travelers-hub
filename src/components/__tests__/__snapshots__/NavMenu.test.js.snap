// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="last:border-l h-auto py-0 border-gray-400 px-2"
  >
    <a
      aria-current="page"
      class="text-blue-600 underline"
      href="/"
    >
      Home
    </a>
  </li>
</DocumentFragment>
`;
